kernel/src/engine/arrow_utils.rs

```diff
@@ -728,11 +731,12 @@ fn parse_json_impl(json_strings: &StringArray, schema: ArrowSchemaRef) -> DeltaR
 
 /// serialize an arrow RecordBatch to a JSON string by appending to a buffer.
 // TODO (zach): this should stream data to the JSON writer and output an iterator.
+#[internal_api]
 pub(crate) fn to_json_bytes(
     data: impl Iterator<Item = DeltaResult<Box<dyn EngineData>>> + Send,
 ) -> DeltaResult<Vec<u8>> {
     let mut writer = LineDelimitedWriter::new(Vec::new());
-    for chunk in data.into_iter() {
+    for chunk in data {
```

Comment:
this is just a drive-by, removing the redundant `into_iter`. surprised clippy did not complain.